flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesClusterDescriptor.java

```diff
@@ -84,6 +86,11 @@ public String getClusterDescription() {
 			if (restEndpoint != null) {
 				configuration.setString(RestOptions.ADDRESS, restEndpoint.getAddress());
 				configuration.setInteger(RestOptions.PORT, restEndpoint.getPort());
+				// Flink client will not use HA service to retrieve address of jobmanager. Instead, it always use Kubernetes
+				// service to contact with jobmanager via rest client.
+				if (HighAvailabilityMode.isHighAvailabilityModeActivated(flinkConfig)) {
```

Comment:
This might work. However, the way is a bit hack and add some dependency/assumption to the behavior of non-HA mode.
Is there a clean way we let the client retrieve jobmanager address from k8s service no matter what the HA mode is?

Reply:
I do not think it is a temporary hack. 
In Kubernetes deployment, when we want to get a ClusterClient, we should not retrieve it from the `ClientHAService`. Instead, we retrieve the rest endpoint from Kubernetes service. 

Maybe removing the condition `if (HighAvailabilityMode.isHighAvailabilityModeActivated(flinkConfig))` will look better. 

```
final Endpoint restEndpoint = client.getRestEndpoint(clusterId);
if (restEndpoint != null) {
  configuration.setString(RestOptions.ADDRESS, restEndpoint.getAddress());
  configuration.setInteger(RestOptions.PORT, restEndpoint.getPort());
}
```

Reply:
I have a serious consideration and will find a more clean way. Maybe adding a new construct method for `RestClusterClient` could work.